src/cases/containers/query-management-container/query-management-container.component.ts

```diff
@@ -1,40 +1,63 @@
+import { ActivatedRoute, Router } from '@angular/router';
+import { CaseNotifier, CaseView, Document, FormDocument, QueryItemType, QueryListItem, partyMessagesMockData } from '@hmcts/ccd-case-ui-toolkit';
+import { FeatureToggleService } from '@hmcts/rpx-xui-common-lib';
+import { Observable, Subscription, combineLatest } from 'rxjs';
+import { map } from 'rxjs/operators';
 import { ErrorMessage } from '../../../app/models/error-message.model';
+import { CaseTypeQualifyingQuestions } from '../../models/qualifying-questions/casetype-qualifying-questions.model';
+import { QualifyingQuestion } from '../../models/qualifying-questions/qualifying-question.model';
 import { RaiseQueryErrorMessage } from '../../models/raise-query-error-message.enum';
 
 @Component({
   selector: 'exui-query-management-container',
   templateUrl: './query-management-container.component.html',
   styleUrls: ['./query-management-container.component.scss']
 })
-export class QueryManagementContainerComponent implements OnInit {
+export class QueryManagementContainerComponent implements OnInit, OnDestroy {
+  private readonly LD_QUALIFYING_QUESTIONS = 'qm-qualifying-questions';
+
+  public caseId: string;
   public queryItem: QueryListItem | undefined;
+  public queryItemType = QueryItemType;
   public showSummary: boolean = false;
   public formGroup: FormGroup = new FormGroup({});
   public submitted = false;
   public errorMessages: ErrorMessage[] = [];
   public queryCreateContext: QueryItemType;
+  public qualifyingQuestions$: Observable<QualifyingQuestion[]>;
+  public qualifyingQuestionsSubscription: Subscription;
 
-  constructor(private activatedRoute: ActivatedRoute) {}
+  constructor(private readonly activatedRoute: ActivatedRoute,
+              private readonly router: Router,
+              private readonly caseNotifier: CaseNotifier,
+              private readonly featureToggleService: FeatureToggleService) {}
 
   public ngOnInit(): void {
-    this.formGroup = new FormGroup({
-      fullName: new FormControl(null, Validators.required),
-      subject: new FormControl(null, Validators.required),
-      body: new FormControl(null, Validators.required),
-      isHearingRelated: new FormControl(null, Validators.required),
-      hearingDate: new FormControl(null),
-      attachments: new FormControl([] as Document[])
-    });
-
     const queryItemId = this.activatedRoute.snapshot.params.qid;
+    this.queryCreateContext = this.getQueryCreateContext(queryItemId);
+
     if (queryItemId) {
       this.queryItem = new QueryListItem();
       Object.assign(this.queryItem, partyMessagesMockData[0].partyMessages[0]);
-      this.queryCreateContext = queryItemId === '1' ? QueryItemType.RESPOND : QueryItemType.FOLLOWUP;
+    }
+
+    if (this.queryCreateContext === QueryItemType.NONE) {
+      this.formGroup = new FormGroup({
+        qualifyingQuestionOption: new FormControl(null, Validators.required)
```

Comment:
Just optionally; I think we should create a formControl (i.e. qualifyingQuestionsControl) and pass that to the component instead of mixing the use of the same property (i.e. formGroup) for both steps?
as it is just one control; it seems like you don't even need a formGroup at all.

Reply:
I am handling the navigation as part of the next ticket https://tools.hmcts.net/jira/browse/EUI-8524, and need the value of the selected radio button to decide.
https://tools.hmcts.net/jira/browse/EUI-8524 is coming out well, and hope all should be in place when I complete the dev work.